--- /Users/lori/Desktop/sunshine/src/stream.h	2025-03-17 11:18:59
+++ /Users/lori/Desktop/nightmare-space/sunshine_android/android/app/src/main/cpp/stream.h	2025-03-16 06:41:13
@@ -14,6 +14,7 @@
 #include "audio.h"
 #include "crypto.h"
 #include "video.h"
+#include "rtsp.h"
 
 namespace stream {
   constexpr auto VIDEO_STREAM_PORT = 9;
\ No newline at end of file
@@ -38,6 +39,8 @@
     std::optional<int> gcmap;
   };
 
+  void postFrame(std::vector<uint8_t> &&frame_data, int64_t frame_index, bool idr, void* channel_data);
+
   namespace session {
     enum class state_e : int {
       STOPPED,  ///< The session is stopped
\ No newline at end of file
@@ -52,4 +55,4 @@
     void join(session_t &session);
     state_e state(session_t &session);
   }  // namespace session
-}  // namespace stream
+}  // namespace stream
\ No newline at end of file
