--- android/sunshine/src/video_colorspace.cpp	2025-03-18 12:40:58
+++ /Users/lori/Desktop/nightmare-space/sunshine_android/android/app/src/main/cpp/video_colorspace.cpp	2025-03-17 13:36:10
@@ -178,6 +178,8 @@
     return result;
   }
 
+#ifndef __ANDROID__  // for compile
+
   const color_t *new_color_vectors_from_colorspace(const sunshine_colorspace_t &colorspace) {
     constexpr auto generate_color_vectors = [](const sunshine_colorspace_t &colorspace) -> color_t {
       double Kr, Kb;
@@ -285,4 +287,5 @@
 
     return result;
   }
+#endif
 }  // namespace video
